### DISCUSSION TITLE
Serverless Hosting

### DISCUSSION THREAD

**Discussion by pspeter3** [pocketbase/pocketbase#4848] [Q&A]
Would it be possible to use [Vercel](https://vercel.com/docs/functions/runtimes/go) and [LibSQL](https://github.com/tursodatabase/go-libsql) to have a serverless hosting?

I think you could export a handler with Go library to meet Vercel's requirements. Is it possible to also dependency inject a SQLite driver?

**Comment by ganigeorgiev** [comment]
> Would it be possible to use [Vercel](https://vercel.com/docs/functions/runtimes/go) and [LibSQL](https://github.com/tursodatabase/go-libsql) to have a serverless hosting?

No. PocketBase requires a single long running process no matter of what database driver you use.

> Is it possible to also dependency inject a SQLite driver?

I'm not sure if that is what you are asking but with the [refactoring](https://github.com/pocketbase/pocketbase/discussions/4355) there will be an option specify custom db connect function and any driver compatible with the std Go `database/sql` and the SQLite SQL dialect should work.

**Reply by pspeter3** [reply]
Got it, thanks! Why does PocketBase require the long running process? Is this for the live updating feature?

**Reply by ganigeorgiev** [reply]
Because there is no builtin synchronization mechanism between multiple running app processes. All registered [app event hooks](https://pocketbase.io/docs/go-event-hooks/) (including those for the realtime APIs) are kept in memory as part of the bootstrapped application instance.
